tests/validator/validator.cpp

```diff
@@ -334,32 +334,71 @@ TEST(Validator, importComponents)
     v.validateModel(m);
     EXPECT_EQ(size_t(0), v.errorCount());
 
-    // Invalid component import- missing refs
+    // Another valid component import
     libcellml::ImportSourcePtr imp2 = std::make_shared<libcellml::ImportSource>();
+    imp2->setUrl("yet-another-other-model.xml");
     libcellml::ComponentPtr importedComponent2 = std::make_shared<libcellml::Component>();
-    importedComponent2->setName("invalid_imported_component_in_this_model");
-    importedComponent2->setSourceComponent(imp2, "");
+    importedComponent2->setName("another_valid_imported_component_in_this_model");
+    importedComponent2->setSourceComponent(imp2, "new_shiny_component_ref");
     m->addComponent(importedComponent2);
     v.validateModel(m);
-    EXPECT_EQ(size_t(3), v.errorCount());
+    EXPECT_EQ(size_t(0), v.errorCount());
 
-    // Invalid component import - duplicate refs
+    // Invalid component import - missing refs
     libcellml::ImportSourcePtr imp3 = std::make_shared<libcellml::ImportSource>();
-    imp3->setUrl("some-other-model.xml");
     libcellml::ComponentPtr importedComponent3 = std::make_shared<libcellml::Component>();
-    importedComponent3->setName("duplicate_imported_component_in_this_model");
-    importedComponent3->setSourceComponent(imp3, "component_in_that_model");
+    importedComponent3->setName("invalid_imported_component_in_this_model");
+    importedComponent3->setSourceComponent(imp3, "");
     m->addComponent(importedComponent3);
     v.validateModel(m);
-    EXPECT_EQ(size_t(4), v.errorCount());
+    EXPECT_EQ(size_t(3), v.errorCount());
 
-    // Invalid component import - unnamed component
+    // Invalid component import - duplicate refs
```

Comment:
This is valid, right? the error count doesn't increase...

Reply:
Sorry - comments were leftovers from previous tests.  You're right, it's permitted to import from the same place.